### PR TITLE
Migrate commits from proposal-temporal 2023-04-12

### PR DESCRIPTION
**Pull request by justingrant**
Porting my first batch of commits from proposal-temporal using @12wrigja's [awesome rebase tool](https://github.com/js-temporal/temporal-polyfill/blob/main/tools/rebasing.md) for porting commits from proposal-temporal repo.

Lots of feedback on the tool is in #236.

If this PR looks good then I can migrate more commits later this week. 

I fixed up a mistake I made in one commit and in the process lost @ptomato's authorship of it. Whoops, sorry Philip!

### CODE REVIEW

**Review by 12wrigja — APPROVED** [review]

File: lib/ecmascript.ts [hunk 1]
```diff
@@ -5959,11 +5955,13 @@ export function CompareISODate(y1: number, m1: number, d1: number, y2: number, m
   return 0;
 }
 
-function NonNegativeModulo(x: number, y: number) {
-  let result = x % y;
-  if (ObjectIs(result, -0)) return 0;
-  if (result < 0) result += y;
-  return result;
+function NonNegativeBigIntDivmod(x: JSBI, y: JSBI) {
+  let { quotient, remainder } = divmod(x, y);
```
Comment (ptomato) [inline lib/ecmascript.ts:5969]: Are you sure that our JSBI `divmod` function has the same semantics as `bigInteger.divmod`? (Just checking; I have no idea)

**Review by ptomato — APPROVED** [review]

File: lib/ecmascript.ts (hunk 1 above)
Comment (justingrant) [reply lib/ecmascript.ts:5969]: Hmmm, I sure hope so given how much we use divmod in this polyfill. @12wrigja you're the JSBI expert, do you know?

Comment (justingrant) [reply lib/ecmascript.ts:5969]: Actually, I wrote a quick test to confirm that results are identical. https://codesandbox.io/s/jsbi-divmod-test-v9zbbn?file=/src/index.js

So I'll go ahead and merge tomorrow morning.

Comment (12wrigja) [reply lib/ecmascript.ts:5969]: I had planned to do the same thing today - I vaguely recall having done something like that when I first needed to reimplement bigInteger.divmod, but didn't have the script or results handy.

Is that script something long-lived that we could link to in a comment so we can easily refer back to it?

**Pull request merged**